File: src/main/java/nnglebanov/daoexample/rest/dto/AuthorDto.java
package nnglebanov.daoexample.rest.dto;

import lombok.Data;
import nnglebanov.daoexample.domain.Author;

@Data
public class AuthorDto {
    private Integer id;
    private String firstName;
    private String lastName;

    public static AuthorDto toDto(Author author) {
        AuthorDto authorDto = new AuthorDto();
        authorDto.setId(author.getId());
        authorDto.setFirstName(author.getFirstName());
        authorDto.setLastName(author.getLastName());
        return authorDto;
    }

    public static Author toEntity(AuthorDto authorDto){
        Author author = new Author();
        author.setFirstName(authorDto.firstName);
        author.setLastName(authorDto.lastName);
        return author;
    }
}

File: src/main/java/nnglebanov/daoexample/rest/dto/AuthorDto.kt
package nnglebanov.daoexample.rest.dto

import nnglebanov.daoexample.domain.Author
import java.util.*


class AuthorDto {
    var id: String? = null
    var firstName: String? = null
    var lastName: String? = null

    companion object {

        fun toDto(author: Author): AuthorDto {
            return AuthorDto().also {
                it.id = author.id
                it.firstName = author.firstName
                it.lastName = author.lastName
            }
        }

        fun toEntity(authorDto: AuthorDto): Author {
            return Author(
                    firstName = authorDto.firstName,
                    lastName = authorDto.lastName,
                    books = null,
                    dateTime = Date()
            )
        }
    }
}
